app/descriptor/views.py

```diff
@@ -93,15 +93,15 @@ def edit_name(desc_id):
     old_name = descriptor.name
     form = EditDescriptorNameForm()
     if form.validate_on_submit():
-        descriptor.name = form.name.data
         if Descriptor.query.filter(Descriptor.name == form.name.data).first() \
                 is not None:
             flash('Descriptor {} already exists.'.format(descriptor.name),
```

Comment:
since we don't change descriptor name to form name until after, we should print `form.name.data` instead of `descriptor.name`